### PR TITLE
Add article about shallow component tests

### PR DESCRIPTION
**Pull request by tispBe**

### CODE REVIEW

**Review by lehmamic — APPROVED** [review]
Nice article, I learned something new :-)

File: drafts/cypress-shallow-component-tests.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+In frontend applications, the emphasis lies on numerous tests validating our component classes and 
+services, with only a sparse selection of e2e tests directly engaging with the UI. 
+However, a component is more than just its class. While class-only tests confirm behavior, 
+they may fall short in assessing interactions with the DOM, and even fail to guarantee accurate rendering.
+
+Angular 17 releases new built-in control flows transferring more logic to the component templates.
+Consequently, the urge to validate component templates and interactions is on the rise, 
+potentially surpassing the significance of unit tests.
+
+Fortunately, Cypress offers a solution in the form of "component tests," applicable to Angular, React, Vue, and Svelte.
```
Comment (romanutti) [inline drafts/cypress-shallow-component-tests.md:22]: "component tests",

File: drafts/cypress-shallow-component-tests.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+          RouterTestingModule,
+          HttpClientTestingModule,
+      ], 
+      providers: [{ provide: MyService, useClass: MyMockService }]
+    });
+});
+```
+Through a combination of reflection, inheritance, and the capabilities of Angular's `TestBed`, 
+we successfully mock child components in standalone component imports. 
+The implementation of our custom mock command has significantly streamlined our component tests.
```
Comment (romanutti) [inline drafts/cypress-shallow-component-tests.md:155]: Can confirm 👍🏼 😄

File: drafts/cypress-shallow-component-tests.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      providers: [{ provide: MyService, useClass: MyMockService }]
+  });
+});
+```
+By leveraging the `MountConfig`, mocking providers and components becomes straightforward using your own 
+MockClass. However, in the context of Angular standalone components, where imports and providers are 
+integral part of the component itself, overriding child components via `MountConfig` encounters limitations.
+
+### Overriding Standalone Child Component
+The easiest solution involves mocking components with third-party libraries like `ng-mocks`.
```
Comment (romanutti) [inline drafts/cypress-shallow-component-tests.md:51]: Did you actively decide against giving an example here? Would be the fast-track for a lot of other projects I guess

**Review by romanutti — APPROVED** [review]
👏🏼

**Pull request merged**